[PREGEL] Add loading state (#1057)
arangodb/docs: 3.10/graphs-pregel.md

| State          | Description    |
|:---------------|:---------------|
| `"running"`    | Algorithm is executing normally.

> | `"in error"`   | The execution is in an error state. This can be caused by primary DB-Servers being not reachable or being non responsive. The execution might recover later, or switch to "canceled" if it was not able to recover successfully

Reviewer: @Simran-B Julia has added three more states (loading, fatal_error, none). Looks like they were removed during your review. Please add them back.

Reviewer 2: Sorry @jvolmer, the visual 3-way merge in VS Code confused me big time. Restored.

| State          | Description    |
|:---------------|:---------------|
| `"none"`       | The pregel run did not yet start
| `"loading"`    | The graph is loaded from the database into memory before the execution of the algorithm
| `"running"`    | Algorithm is executing normally.
| `"storing"`    | The algorithm finished, but the results are still being written back into the collections. Occurs if the `store` parameter is set to `true` only.
| `"done"`       | The execution is done. In version 3.7.1 and later, this means that storing is also done. In earlier versions, the results may not be written back into the collections yet. This event is announced in the server log (requires at least *info* log level for the *pregel* topic).
| `"canceled"`   | The execution was permanently canceled, either by the user or by an error.
| `"in error"`   | The execution is in an error state. This can be caused by primary DB-Servers being not reachable or being non responsive. The execution might recover later, or switch to "canceled" if it was not able to recover successfully

> | `"recovering"` | The execution is actively recovering, will switch back to "running" if the recovery was successful

Reviewer 2: ```suggestion
| `"recovering"` | The execution is actively recovering and switches back to "running" if the recovery is successful
```
